### PR TITLE
Xamarin LiteServ Android Support

### PR DESCRIPTION
**Pull request by borrrden**
- [X] Ran `flake8`

#### Changes proposed in this pull request:

- Add Xamarin Android as an endpoint for functional testing

### CODE REVIEW

File: keywords/LiteServAndroid.py [hunk 1]
```diff
@@ -1,83 +1,41 @@
...
-        apk_path = "{}/{}".format(BINARY_DIR, apk_name)
         log_info("Installing: {}".format(apk_path))
 
-        output = subprocess.check_output(["adb", "install", apk_path])
-        if "INSTALL_FAILED_ALREADY_EXISTS" in output or "INSTALL_FAILED_UPDATE_INCOMPATIBLE" in output:
-            raise LiteServError("Error. APK already exists.")
+        process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
+        output, _ = process.communicate()
+        process.poll()
+        if "INSTALL_FAILED_ALREADY_EXISTS" in output or \
```
Comment (raghusarangapani) [inline keywords/LiteServAndroid.py]: Need a quick clarification. From my testing, adb prints INSTALL_FAILED_ALREADY_EXISTS or INSTALL_FAILED_UPDATE_INCOMPATIBLE on the terminal console. Will output, _ trick grab the console output into "output"?

File: keywords/LiteServiOS.py [hunk 2]
```diff
@@ -32,8 +61,8 @@ def install(self):
         if self.storage_engine != "SQLite":
             raise LiteServError("https://github.com/couchbaselabs/liteserv-ios/issues/1")
 
-        package_name = "couchbase-lite-ios-liteserv-{}.app".format(self.version_build)
-        app_path = "{}/{}".format(BINARY_DIR, package_name)
+        package_name = "LiteServ-iOS.app"
+        app_path = "{}/{}/{}".format(BINARY_DIR, "LiteServ-iOS", package_name)
```
Comment (raghusarangapani) [inline keywords/LiteServiOS.py]: Quick clarification. The new code in LiteServiOS.py was already in the branch right? The LiteServ-iOS.app currently being built is only for emulator. I and Seth had discussed adding some logic to detect the device and continue with emulator if no device is found. In that case, the host and a few other variables have to be reset. You can implement what you think is best.

File: keywords/LiteServAndroid.py (hunk 1 above)
Comment (borrrden) [inline keywords/LiteServAndroid.py]: That is my understanding. The code before was simply raising an exception.

Comment (raghusarangapani) [inline keywords/LiteServAndroid.py]: >>> import subprocess
>>> apk_path
'/Users/raghu.sarangapani/Downloads/LiteServ/Android/LiteServ.apk'
>>> process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
>>> output, _ = process.communicate()
Success
>>> output
''
>>> process = subprocess.Popen(["adb", "install", apk_path], stdout=subprocess.PIPE)
>>> output, _ = process.communicate()
Failed to install /Users/raghu.sarangapani/Downloads/LiteServ/Android/LiteServ.apk: Failure [INSTALL_FAILED_ALREADY_EXISTS: Attempt to re-install com.couchbase.liteserv without first uninstalling.]
>>> output
''

The output from adb is not captured in "output" variable.

Comment (borrrden) [inline keywords/LiteServAndroid.py]: Oh, it might be stderr instead!  I will capture both just in case.

File: keywords/LiteServJavaAndroid.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            # to make sure the dbs are encrypted during the tests
+            db_flags = []
+            for db_name in REGISTERED_CLIENT_DBS:
+                db_flags.append("{}:pass".format(db_name))
+            db_flags = ",".join(db_flags)
+
+            log_info("Running with db_flags: {}".format(db_flags))
+
+            if self.storage_engine == "SQLCipher":
+                output = subprocess.check_output([
```
Comment (raghusarangapani) [inline keywords/LiteServJavaAndroid.py]: Can we use launch_args and launch_args.extend as in https://github.com/couchbaselabs/mobile-testkit/pull/988/files

File: keywords/LiteServJavaAndroid.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    "--es", "password", "none",
+                    "--ei", "listen_port", str(self.port),
+                    "--es", "storage", "ForestDB",
+                    "--es", "dbpassword", db_flags
+                ])
+                log_info(output)
+        else:
+            log_info("No encryption ...")
+            output = subprocess.check_output([
+                "adb", "shell", "am", "start", "-n", self.activity_name,
```
Comment (raghusarangapani) [inline keywords/LiteServJavaAndroid.py]: Can we use launch_args and launch_args.extend as in https://github.com/couchbaselabs/mobile-testkit/pull/988/files

**Comment by raghusarangapani** [comment]
Can you also kick off a run on jenkins with this branch just to make sure nothing is broken?

File: keywords/LiteServiOS.py (hunk 2 above)
Comment (sethrosetter) [inline keywords/LiteServiOS.py]: This will be fine for now. I think we will have to fix a few things when iOS is building in CI

**Comment by sethrosetter** [comment]
Test run in progress - http://uberjenkins.sc.couchbase.com/job/android-listener-client-sync-gateway/325/

**Pull request closed without merging**